pkg/ring/ring.go

```diff
@@ -396,3 +397,56 @@ func (r *Ring) Collect(ch chan<- prometheus.Metric) {
 		r.name,
 	)
 }
+
+// Subring returns a ring of n ingesters from the given ring
+// Subrings are meant only for ingestor lookup and should have their data externalized.
```

Comment:
You could enforce it a bit returning a `ReadRing` instead of `*Ring`

Reply:
Since I have no reason for restricting behaviour I think it's better practice to return the full struct.

Reply:
What we're returning here is not a fully functional ring (ie. created with `ring.New()`). By fully functional, I mean a ring honoring the ring's features contract, that's expected. We're just returning a sub-view of the ring, so restricting the behaviour returning with `ReadRing` looks more appropriate to me.

Reply:
That's a good point. Will change.

Reply:
Fixed.